pandas/_libs/lib.pyx

```diff
@@ -1100,6 +1103,8 @@ cdef inline bint c_is_list_like(object obj, bint allow_sets) except -1:
         and not (hasattr(obj, "ndim") and obj.ndim == 0)
         # exclude sets if allow_sets is False
         and not (allow_sets is False and isinstance(obj, abc.Set))
+        # exclude Generic types that have __iter__
+        and not isinstance(obj, _GenericAlias)
```

Comment:
Coul also combine it with the earlier exception for `str` and `bytes`

Reply:
Is there a public class to test against?

Reply:
> Coul also combine it with the earlier exception for `str` and `bytes`

done in https://github.com/pandas-dev/pandas/pull/49736/commits/5260e4cff5076c6aaeb4883a0835988e74d2a7e3

Reply:
Can we simply test against `type`: `not isinstance(obj, type)`? Not sure whether that might be too wide.

Reply:
> Can we simply test against `type`: `not isinstance(obj, type)`? Not sure whether that might be too wide.

No.  See below.
```python
>>> from typing import TypeVar, Generic
>>> T = TypeVar("T")
>>> class MyDataFrame(DataFrame, Generic[T]):
...     ...
...
>>> tstc = MyDataFrame[int]
>>> isinstance(tstc, type)
False
>>> from typing import _GenericAlias
>>> isinstance(tstc, _GenericAlias)
True
```